common/service/config/config.go

```diff
@@ -237,6 +237,14 @@ type (
 		FlushBytes int `yaml:"flushBytes"`
 	}
 
+	// Archival contains the config for archival
+	Archival struct {
+		// EnableArchival whether archival is enabled
+		EnableArchival bool `yaml:"enableArchival"`
+		// Blobstore the configuration for blobstore
+		Blobstore filestore.Config `yaml:"blobstore"`
```

Comment:
- can we call this filestore instead of blobstore so that we can add additional config item for each store type (s3, gs, azureblobstore etc) in future ? I don' imagine specifying storeDirectory or bucket retention / ownership for non-filestore based options
- optionally, you can also consider having a storeType enum {filestore, s3, gs... }